docs/src/docs/asciidoc/configuration-model.adoc

```diff
@@ -3,22 +3,231 @@
+----
+
+[NOTE]
+`ProviderSettings` is a *REQUIRED* component.
+
+[TIP]
+<<configure-defaults, `@Import(OAuth2AuthorizationServerConfiguration.class)`>> automatically registers a `ProviderSettings` `@Bean`, if not already provided.
+
+The following example shows how to customize the configuration settings and register a `ProviderSettings` `@Bean`:
+
+[source,java]
+----
+@Bean
+public ProviderSettings providerSettings() {
+	return ProviderSettings.builder()
+		.issuer("https://example.com")
+		.authorizationEndpoint("/oauth2/v1/authorize")
+		.tokenEndpoint("/oauth2/v1/token")
+		.tokenIntrospectionEndpoint("/oauth2/v1/introspect")
+		.tokenRevocationEndpoint("/oauth2/v1/revoke")
+		.jwkSetEndpoint("/oauth2/v1/jwks")
+		.oidcUserInfoEndpoint("/connect/v1/userinfo")
+		.oidcClientRegistrationEndpoint("/connect/v1/register")
+		.build();
+}
+----
+
+The `ProviderContext` is a context object that holds information about the provider.
+It provides access to the `ProviderSettings` and the "current" issuer identifier.
+
+[NOTE]
+If the issuer identifier is not configured in `ProviderSettings.builder().issuer(String)`, it will be resolved from the current request.
+
+The `ProviderContext` is accessible through the `ProviderContextHolder`, which associates it with the current request thread using a `ThreadLocal`.
+
+[NOTE]
+The `ProviderContextFilter` associates the `ProviderContext` with the `ProviderContextHolder`.
+
+[[oauth2-client-authentication]]
+== Configuring Client Authentication
+
+`OAuth2ClientAuthenticationConfigurer` provides the ability to customize https://datatracker.ietf.org/doc/html/rfc6749#section-2.3[OAuth2 client authentication].
+It defines extension points that allow you to customize the pre-processing, main processing, and post-processing logic for client authentication requests.
+
+`OAuth2ClientAuthenticationConfigurer` provides the following configuration options:
+
+[source,java]
+----
+@Bean
+public SecurityFilterChain authorizationServerSecurityFilterChain(HttpSecurity http) throws Exception {
+	OAuth2AuthorizationServerConfigurer<HttpSecurity> authorizationServerConfigurer =
+		new OAuth2AuthorizationServerConfigurer<>();
+	http.apply(authorizationServerConfigurer);
+
+	authorizationServerConfigurer
+		.clientAuthentication(clientAuthentication ->
+			clientAuthentication
+				.authenticationConverter(authenticationConverter)   <1>
+				.authenticationProvider(authenticationProvider) <2>
+				.authenticationSuccessHandler(authenticationSuccessHandler) <3>
+				.errorResponseHandler(errorResponseHandler) <4>
+		);
+
+	return http.build();
+}
+----
+<1> `authenticationConverter()`: The `AuthenticationConverter` (_pre-processor_) used when attempting to extract client credentials from `HttpServletRequest` to an instance of `OAuth2ClientAuthenticationToken`.
+<2> `authenticationProvider()`: The `AuthenticationProvider` (_main processor_) used for authenticating the `OAuth2ClientAuthenticationToken`. (one or more may be added)
```

Comment:
Should we indicate that "(one or more may be added *to replace the defaults*)" or similar? (since there have been conversations/questions about overriding default providers a few times)

Reply:
Change `(one or more may be added)` to `(One or more may be added.)`. (It is a sentence.)